docs/migration_guides/opflow_migration.rst

```diff
@@ -0,0 +1,1156 @@
+     - :class:`~qiskit.quantum_info.Statevector` or simply :class:`~qiskit.QuantumCircuit`, depending on the use case.
+
+       ..  note::
+
+           For efficient simulation of stabilizer states, ``quantum_info`` includes a :class:`~qiskit.quantum_info.StabilizerState` class. See API ref. for more info.
+
+Example 1
+##########
+.. raw:: html
+
+    <details>
+    <summary><a><b>Opflow</b></a></summary>
+
+.. code-block:: python
+
+    from qiskit.opflow import Zero, One, Plus, Minus
+
+    # Zero, One, Plus, Minus are all stabilizer states
+    state1 = Zero ^ One
+    state2 = Plus ^ Minus
+
+.. raw:: html
+
+   </details>
+
+.. raw:: html
+
+    <details>
+    <summary><a><b>Alternative</b></a></summary>
+
+.. code-block:: python
+
+    from qiskit import QuantumCircuit
+    from qiskit.quantum_info import StabilizerState, Statevector
+
+    qc_zero = QuantumCircuit(1)
+    qc_one = qc_zero.copy()
+    qc_one.x(0)
+    state1 = Statevector(qc_zero) ^ Statevector(qc_one)
+
+    qc_plus = qc_zero.copy()
+    qc_plus.h(0)
+    qc_minus = qc_one.copy()
+    qc_minus.h(0)
+
+    state2 = StabilizerState(qc_plus) ^ StabilizerState(qc_minus)
+
+
+.. raw:: html
+
+   </details>
+
+
+Primitive and List Ops
+----------------------
+*Back to* `Index`_
+
+Most of the workflows that previously relied in components from :mod:`~qiskit.opflow.primitive_ops` and
+:mod:`~qiskit.opflow.list_ops` can now leverage elements from ``quantum_info``\'s :mod:`~qiskit.quantum_info.operators` instead.
+Some of these classes do not require a 1-1 replacement because they were created to interface with other
+opflow components.
+
+Primitive Ops
+~~~~~~~~~~~~~~
+*Back to* `Index`_
+
+:class:`~qiskit.opflow.primitive_ops.PrimitiveOp` is the :mod:`~qiskit.opflow.primitive_ops` module's base class.
+It also acts as a factory to instantiate a corresponding sub-class depending on the computational primitive used
+to initialize it.
+
+.. tip::
+
+    Interpreting :class:`~qiskit.opflow.primitive_ops.PrimitiveOp` as a factory class:
+
+    .. list-table::
+       :header-rows: 1
+
+       * - Class passed to :class:`~qiskit.opflow.primitive_ops.PrimitiveOp`
+         - Sub-class returned
+
+       * - :class:`~qiskit.quantum_info.Pauli`
+         - :class:`~qiskit.opflow.primitive_ops.PauliOp`
+
+       * - :class:`~qiskit.circuit.Instruction`, :class:`~qiskit.circuit.QuantumCircuit`
+         - :class:`~qiskit.opflow.primitive_ops.CircuitOp`
+
+       * - ``list``, ``np.ndarray``, ``scipy.sparse.spmatrix``, :class:`~qiskit.quantum_info.Operator`
+         - :class:`~qiskit.opflow.primitive_ops.MatrixOp`
+
+Thus, when migrating opflow code, it is important to look for alternatives to replace the specific subclasses that
+might have been used "under the hood" in the original code:
+
+.. list-table::
+   :header-rows: 1
+
+   * - Opflow
+     - Alternative
+
+   * - :class:`~qiskit.opflow.primitive_ops.PrimitiveOp`
+     - :class:`~qiskit.quantum_info.Operator`
```

Comment:
As you explained nicely above, the `PrimitiveOp` was more a factory and base class, so I wouldn't say that `Operator` is a replacement as that strictly stores matrix representations 🙂 I think this just doesn't have (or need) an alternative